src/mod/features/battle/battle_situation.cpp

```diff
@@ -0,0 +1,504 @@
+        __this->fields._IsShow_k__BackingField = true;
+        __this->fields._animationState_k__BackingField = 2;
+
+        auto uiCanvasBase = (Dpr::Battle::View::UI::BattleViewUICanvasBase::Object*)__this;
+        uiCanvasBase->SelectButton(__this->fields._situationButtons, __this->fields._CurrentIndex_k__BackingField, false, Dpr::Battle::View::UI::BattleViewUICanvasBase::Method$$BUISituationButton$$SelectButton);
+
+        Dpr::Battle::View::BattleViewCore::getClass()->initIfNeeded();
+        auto battleViewCore = Dpr::Battle::View::BattleViewCore::get_Instance();
+
+        battleViewCore->fields._UISystem_k__BackingField->fields._cursor->SetActive(true);
+    }
+};
+
+HOOK_DEFINE_REPLACE(BUISituation_OnSubmit) {
+    static void Callback(Dpr::Battle::View::UI::BUISituation::Object* __this) {
+        Logger::log("BUISituation_OnSubmit\n");
+
+        system_load_typeinfo(0x1f51);
+
+        Dpr::Battle::View::BattleViewCore::getClass()->initIfNeeded();
+        Dpr::Battle::View::Systems::BattleViewUISystem::Object* battleViewUISystem = Dpr::Battle::View::BattleViewCore::get_Instance()->fields._UISystem_k__BackingField;
+
+        __this->fields._IsValid_k__BackingField = true;
+        __this->fields._IsFocus_k__BackingField = false;
+
+        auto mons = __this->fields._monsParams->get_Values<System::Collections::Generic::IEnumerable$$BTL_POKEPARAM>();
+        auto monsList = System::Linq::Enumerable::ToList<System::Collections::Generic::IEnumerable$$BTL_POKEPARAM, System::Collections::Generic::List$$BTL_POKEPARAM, Dpr::Battle::Logic::BTL_POKEPARAM>(mons->instance());
+
+        auto trainerNames = __this->fields._trainerNames->get_Values<System::Collections::Generic::IEnumerable$$String>();
+        auto trainerNamesList = System::Linq::Enumerable::ToList<System::Collections::Generic::IEnumerable$$String, System::Collections::Generic::List$$String, System::String>(trainerNames->instance());
+
+        auto icons = __this->fields._icons->get_Values<System::Collections::Generic::IEnumerable$$UnityEngine_UI_Image>();
+        auto iconsList = System::Linq::Enumerable::ToList<System::Collections::Generic::IEnumerable$$UnityEngine_UI_Image, System::Collections::Generic::List$$UnityEngine_UI_Image, UnityEngine::UI::Image>(icons->instance());
+
+        Dpr::Battle::Logic::BTL_POKEPARAM::Object* currentMon = __this->fields._monsParams->get_Item(__this->fields._CurrentIndex_k__BackingField)->instance();
+        int32_t index = monsList->IndexOf(currentMon);
+
+        ((Dpr::Battle::View::UI::BattleViewUICanvasBase::Object *) __this)->Hide(false, nullptr);
+        auto situationDetailUI = getSituationDetailUI(battleViewUISystem);
+        situationDetailUI->Initialize(monsList->instance(), trainerNamesList->instance(), iconsList->instance(), index);
+        ((Dpr::Battle::View::UI::BattleViewUICanvasBase::Object *) situationDetailUI)->Show(nullptr);
+
+        battleViewUISystem->PlaySe(0x132562f0);
```

Comment:
I feel like we should have the "COMMON_UI" sound effects in a header so we can have consts set up

Reply:
All the sound effects are in AK.Events, we can eventually add that header for all of them